Python/compile.c

```diff
@@ -4918,25 +4948,44 @@ stackdepth_walk(struct compiler *c, basicblock *b, int depth, int maxdepth)
             maxdepth = depth;
         assert(depth >= 0); /* invalid code or bug in stackdepth() */
         if (instr->i_jrel || instr->i_jabs) {
+            /* Recursively inspect jump target */
             target_depth = depth;
+            if (instr->i_opcode == CONTINUE_LOOP) {
+                /* Pops variable number of values from the stack,
+                 * but the target should be already proceeding.
+                 */
+                assert(instr->i_target->b_seen);
+                assert(instr->i_target->b_startdepth <= depth);
+                goto out; /* remaining code is dead */
+            }
             if (instr->i_opcode == FOR_ITER) {
-                target_depth = depth-2;
+                target_depth = depth - 2;
```

Comment:
Not sure why. At best, `FOR_ITER` calls `STACKADJ(-1)` (when the iterator is exhausted).

Reply:
See PyCompile_OpcodeStackEffect(). The effect is +1 in normal flow and -1 when jumps.

Reply:
So where does `-2` come from?

Reply:
`+1 - 2 = -1`